charts/tfy-karpenter-config/values.yaml

```diff
@@ -157,7 +157,108 @@ karpenter:
-    userData: ""
+    userData: |
+      #!/bin/bash
+      set -ex
+
+      version_lte() {
+          printf '%s\n' "$1" "$2" | sort -C -V
+      }
+
+      version_lt() {
+          ! version_lte "$2" "$1"
+      }
+
+      CONTAINERD_VERSION_OUTPUT="$(containerd --version)"
+      IFS=" " read -a  CONTAINERD_VERSION_STRING <<< "${CONTAINERD_VERSION_OUTPUT}"
+      CONTAINERD_VERSION=${CONTAINERD_VERSION_STRING[2]}
+      echo containerd version "${CONTAINERD_VERSION}"
+
+      setup_soci() {( set -ex
+        yum install fuse -y
+        modprobe fuse
+
+        curl \
+        --silent \
+        --show-error \
+        --retry 10 \
+        --retry-delay 1 \
+        -L https://github.com/awslabs/soci-snapshotter/releases/download/v0.4.0/soci-snapshotter-0.4.0-linux-amd64.tar.gz \
+        -o soci-snapshotter-0.4.0-linux-amd64.tar.gz
+
+        tar -C /usr/local/bin -xvf soci-snapshotter-0.4.0-linux-amd64.tar.gz soci soci-snapshotter-grpc
+        rm soci-snapshotter-0.4.0-linux-amd64.tar.gz
+
+        cat > /etc/systemd/system/soci-snapshotter.service << EOF
+      [Unit]
+      Description=soci snapshotter containerd plugin
+      Documentation=https://github.com/awslabs/soci-snapshotter
+      After=network.target
+      Before=containerd.service
+
+      [Service]
+      Type=notify
+      ExecStart=/usr/local/bin/soci-snapshotter-grpc
+      Restart=always
+      RestartSec=5
+
+      [Install]
+      WantedBy=multi-user.target
+      EOF
+
+        systemctl daemon-reload
+        systemctl enable --now soci-snapshotter
+        systemctl status soci-snapshotter
+
+        echo patching /etc/eks/containerd/containerd-config.toml
+        cat > /etc/eks/containerd/containerd-config.toml << EOF
+      version = 2
+      root = "/var/lib/containerd"
+      state = "/run/containerd"
+
+      [grpc]
+      address = "/run/containerd/containerd.sock"
+
+      [plugins."io.containerd.grpc.v1.cri".containerd]
+      default_runtime_name = "runc"
+      discard_unpacked_layers = true
+      ## ENABLE SOCI SNAPSHOTTER
+      snapshotter = "soci"
+      disable_snapshot_annotations = false
+      ##
+
+      [plugins."io.containerd.grpc.v1.cri"]
+      sandbox_image = "602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/pause:3.5"
+
+      [plugins."io.containerd.grpc.v1.cri".registry]
+      config_path = "/etc/containerd/certs.d:/etc/docker/certs.d"
+
+      [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
+      runtime_type = "io.containerd.runc.v2"
+
+      [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
+      SystemdCgroup = true
+
+      [plugins."io.containerd.grpc.v1.cri".cni]
+      bin_dir = "/opt/cni/bin"
+      conf_dir = "/etc/cni/net.d"
+
+      ## SOCI PLUGIN
+      [proxy_plugins]
+      [proxy_plugins.soci]
+      type = "snapshot"
+      address = "/run/soci-snapshotter-grpc/soci-snapshotter-grpc.sock"
+      ##
+      EOF
+
+      systemctl restart containerd
```

Comment:
Is this required? I think containerd is not started at this point

Reply:
Yes, weird. For whatever reason, it was not working without this restart. I will check again.